### PR TITLE
[ALDN-130] Allow running minikube with vm-driver=none

### PR DESCRIPTION
**Pull request by ptramsey**
This PR updates aladdin to allow running natively on the host's docker
 if available, by setting the new minikube.vm_driver preference in
 ~/.aladdin/config/config.json to 'none' (the default is 'virtualbox').
 This also adds a 'minikube.memory' preference to allow configuring
 the minikube VM's memory, if applicable.
 
 [ALDN-130](https://fivestars.atlassian.net/browse/ALDN-130)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -33,6 +34,9 @@ RUN	curl -L -o /bin/kops https://github.com/kubernetes/kops/releases/download/$K
 RUN	curl -L -o- https://storage.googleapis.com/kubernetes-helm/helm-v$HELM_VERSION-linux-amd64.tar.gz | tar -zxvf - && cp linux-amd64/helm \
 	/bin/helm && chmod 755 /bin/helm && helm init --client-only
 
+RUN curl -L -o- https://download.docker.com/linux/static/stable/x86_64/docker-$DOCKER_VERSION.tgz | tar -zxvf - && cp docker/docker \
```
Comment (ptramsey) [inline Dockerfile]: We no longer depend on mounting the docker executable from the underlying system, as there's no guarantee that it will be statically linked (and therefore run inside our alpine/musl-based aladdin container).  Since we already directly download the docker executable in infra_k8s_check.sh, doing it here too seemed reasonable.

File: aladdin.sh [hunk 2]
```diff
@@ -85,22 +89,54 @@ function check_and_handle_init() {
 }
 
 function set_minikube_config(){
+    local vm_driver=$(jq --arg DEFAULT "$DEFAULT_MINIKUBE_VM_DRIVER" -r '.minikube.vm_driver//$DEFAULT' "$ALADDIN_CONFIG_FILE")
+    local memory=$(jq --arg DEFAULT "$DEFAULT_MINIKUBE_MEMORY" -r '.minikube.memory//$DEFAULT' "$ALADDIN_CONFIG_FILE")
```
Comment (agianey) [inline aladdin.sh]: Is the expectation here that if someone wants to use a different vm driver/memory, they need to update their aladdin config? Currently I think the pattern is for users to not really touch their aladdin config, but I'm open to that pattern being changed.

File: aladdin.sh [hunk 3]
```diff
@@ -85,22 +89,54 @@ function check_and_handle_init() {
...
+            minikube_cmd="sudo -E CHANGE_MINIKUBE_NONE_USER=true '$(which minikube)' start"
+
+        else
+            # On linux, /home gets mounted on /hosthome in the minikube vm, so as not to
+            # override /home/docker.  We still want the user's home directory to be in the
+            # same path both in and outside the minikube vm though, so symlink it into place.
+            minikube_cmd="$minikube_cmd &&\
+                          minikube ssh \"sudo mkdir '$HOME' && \
+                                         sudo mount --bind '${HOME/\/home//hosthome}' '$HOME'\""
+        fi
```
Comment (agianey) [inline aladdin.sh]: This part looks fine to me, but maybe beneficial to get @bbil to look at this section?

File: aladdin.sh (hunk 2 above)
Comment (agianey) [reply aladdin.sh]: Maybe add support for cpu and disk size while you're in here? i think that will take care of the "most popularly" toggled minikube settings

Comment (ptramsey) [reply aladdin.sh]: Done.

Comment (ptramsey) [reply aladdin.sh]: Switched to using aladdin config set/get, rather than looking at a specific file.

File: aladdin.sh [hunk 4]
```diff
@@ -86,21 +90,59 @@ function check_and_handle_init() {
 
 function set_minikube_config(){
     minikube config set kubernetes-version v$KUBERNETES_VERSION &> /dev/null
-    minikube config set vm-driver virtualbox &> /dev/null
-    minikube config set memory 4096 &> /dev/null
+
+    for key in vm_driver memory disk_size cpus; do
+        local minikube_key=$(tr _ - <<< "$key")  # e.g., vm-driver
+        local default_var="DEFAULT_MINIKUBE_$(tr a-z A-Z <<< "$key")"  # e.g., DEFAULT_MINIKUBE_VM_DRIVER
```
Comment (agianey) [inline aladdin.sh]: I think we're missing defaults for cpu and disk size?

File: commands/bash/host/config/config [hunk 5]
```diff
@@ -12,19 +12,22 @@ function config() {
...
         set)
-            jq -c --arg key "$1" --arg val "$2" '. + { ($key): $val }' "$config_path" > "tmp.$$.json" && mv "tmp.$$.json" "$config_path"
+            jq --arg key "$1" --arg val "$2" 'setpath($key | split("."); $val)' "$config_path" > "tmp.$$.json" && mv "tmp.$$.json" "$config_path"
         ;;
         view)
             cat "$config_path"
         ;;
         unset)
-            jq -c "del(.$1)" "$config_path" > "tmp.$$.json" && mv "tmp.$$.json" "$config_path"
+            jq --arg key "$1" 'delpaths([$key | split(".")])' "$config_path" > "tmp.$$.json" && mv "tmp.$$.json" "$config_path"
```
Comment (agianey) [inline commands/bash/host/config/config]: some context/reasoning around these changes would be greatly appreciated :)

File: aladdin.sh (hunk 3 above)
Comment (bbil) [reply aladdin.sh]: This is for virtual box right?

This looks like it is replacing the manual steps I used to do whenever creating a new minikube vm (through the virtual box UI).

File: aladdin.sh [hunk 6]
```diff
@@ -86,21 +90,59 @@ function check_and_handle_init() {
...
+
         echo "Starting minikube... (this will take a moment)"
         set_minikube_config
-        minikube start &> /dev/null
+
+        _start_minikube
+
         # Determine if we've installed our bootlocal.sh script to replace the vboxsf mounts with nfs mounts
         if ! "$(minikube ssh -- "test -x /var/lib/boot2docker/bootlocal.sh && echo -n true || echo -n false")"; then
             echo "Installing NFS mounts from host..."
```
Comment (bbil) [inline aladdin.sh]: maybe put this "installing..." message in the condition below

File: commands/bash/host/config/config (hunk 5 above)
Comment (ptramsey) [reply commands/bash/host/config/config]: Basically, we were passing in the key/value as --args to jq during set, but not during get or unset.  As a result, we got different behavior between those two cases.  In set, a dotted key (e.g., minikube.vm_driver) would be set literally --- `"minikube.vm_driver": "virtualbox"` --- but in unset and get, it would look for 
```
"minikube": {
    "vm_driver": "virtualbox"
}
```

Unfortunately, using --arg is safer than directly interpolating the key/value into the string (reduces confusion around what is parsing '$'; prevents injection).

So, to recap:

for get/unset, we were interpolating dotted values directly into the query string, and getting e.g. .minikube.vm_driver.

For set, we were passing in the key as an argument, but then using it wrong (effectively doing .["minikube.vm_driver"]).

This new code always passes values in as arguments, and then uses them correctly from within jq to address their corresponding locations in the hierarchy.

File: aladdin.sh (hunk 4 above)
Comment (ptramsey) [reply aladdin.sh]: Yes.  I'm not specifying them here, and I explain why in the README --- there's no reason to override the default behavior in minikube.

File: aladdin.sh (hunk 3 above)
Comment (ptramsey) [reply aladdin.sh]: yup!

**Comment by agianey** [comment]
Did some smoke testing on mac, and things looked good
